src/sid_list.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Roman Gezikov <rgezikov@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef SRC_SID_LIST_H_
+#define SRC_SID_LIST_H_
+
+#include <libswiftnav/common.h>
+#include <libswiftnav/signal.h>
+
+typedef struct {
+  gnss_signal_t *l;
+  u16 len;
+  u16 max_len;
+} gnss_signal_list_t;
```

Comment:
len is not an index, it's number of elements. Both 'len' and 'max_len' show number of elements in the list.